src/app/ngrx/effects/routing.effects.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { RouterStateSnapshot } from '@angular/router';
+import 'rxjs/add/operator/switchMap';
+import { Observable } from 'rxjs/Observable';
+import { Action } from '@ngrx/store';
+import { ROUTER_NAVIGATION, RouterNavigationPayload, RouterNavigationAction } from '@ngrx/router-store';
+import { Actions, Effect } from '@ngrx/effects';
+import { FilterModel, IntervalList } from '../model';
+import { CastleFilterAction } from '../actions';
+import { getStandardRangeForBeginEndDate, getBeginEndDateFromStandardRange, getRange } from '../../shared/util/date.util';
+
+@Injectable()
+export class RoutingEffects {
+  @Effect()
+  filterFromRoute$: Observable<Action> = this.actions$
+    .ofType(ROUTER_NAVIGATION)
+    .map((action: RouterNavigationAction) => action.payload)
+    .switchMap((payload: RouterNavigationPayload<RouterStateSnapshot>) => {
+      const filter: FilterModel = {};
+      if (payload.routerState.root.firstChild.params &&
+        payload.routerState.url && payload.routerState.url.length > 1) {
+        const { params } = payload.routerState.root.firstChild; // ActivatedRoute params equiv
+        const url = payload.routerState.url.split('/');
+        if (params['seriesId'] && !isNaN(parseInt(params['seriesId'], 10))) {
+          filter.podcastSeriesId = +params['seriesId'];
+        }
+        if (url.length >= 3 && url[2]) {
+          const type = url[2]; // "downloads" MetricsType
```

Comment:
I'm not currently using type, everything is "downloads" on the state, podcast['downloadsDaily'], podcast['downloadsHourly'], podcast['downloadsMonthly'], etc. (same for episodes.) The idea of type though is so that when/if(?) we add geo or user agent/source or cumulative download data, it will use type to manage additional state for those kind of metrics, podcast['geoDaily'] or podcast['sourceDaily']. I think(?) that I'm going to get the go ahead to work on either user agents/sources or geo data soon, but it's hard to say because I haven't been allowed to see the designs nor am I privy to the plans for the project. Making my best guess about the plan, I'm hanging onto type and put it into the route because I do have a plan for it.

Reply:
👍